src/resources/listings.js

```diff
@@ -148,6 +149,25 @@ class Listings extends ResourceBase {
   async create(data, schemaType, confirmationCallback) {
     const listingType = data.listingType || unitListingType
     data.listingType = listingType // in case it wasn't set
+
+    if (data.pictures) {
+      // Filter picture URLs and upload data: URLs to ipfs
```

Comment:
My reasoning here was that we should do all we can to make it easier on the dapp developers. It's simpler to pass the `data:` URIs or just form data rather than doing all the IPFS uploads themselves. It's definitely more complexity for `origin-js` though. I think there are decent arguments in both directions, it really depends how much magic we want in `origin-js`. :man_shrugging:

Reply:
Yes, our goal should be to make it as easy as possible for the DApp developers.
Now, thinking a bit more about it, changing this interface would require a DApp change as a separate PR... I wish we had a mono-repo :)
Alright, let's keep this PR small and leave as is for now - as we clean up the origin-js library we can revisit. 

But perhaps add a comment at the top of the create function that documents that data.pictures items can either be an IPFS URL or the data itself and in which case it gets saved to IPFS.

Reply:
Agreed that the DApp should not even need to know what IPFS is, let alone interact with it before creating listings ✌️

Reply:
Should we add some kind of get image method for origin.js? You don't usually want to be fetching the images with a `marketplace.get`.

At the same time, makes a ton of sense that the DAPP doesn't need to know where the images come from.

Reply:
This PR will make it so that the images aren't retrieved along with the listing. The images will be an array of `http(s):` URLs (stored on IPFS and accessed through the configured IPFS gateway) rather than `data:` URLs as before so there shouldn't be too much additional overhead if they are included.